### PR TITLE
Implemented code coverage reporting using the coverage.py tool

### PR DESCRIPTION
**Pull request by Sekhar-Kumar-Dash**
## Changes proposed
In this Pull Request I have implemented code coverage reporting using the coverage.py tool.

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by Sekhar-Kumar-Dash** [comment]
hey @AlyaGomaa  Thank you for reviewing my PR and thanks for the suggestions. I will make the required changes and let you know once I have done so

**Comment by Sekhar-Kumar-Dash** [comment]
Hey @AlyaGomaa , I have made the required changes to the PR. Please review it and let me know if any further changes are needed.

File: .github/workflows/CI-staging.yml [hunk 1]
```diff
@@ -1,60 +1,192 @@
...
-            # features will be added to this branch using PRs, not need to re-run the tests on push
-            - '!develop'
-            - '!master'
-    pull_request:
-        branches:
-            - 'develop'
-            - '!master'
+  pull_request:
+    branches:
+    - 'master'
```
Comment (AlyaGomaa) [inline .github/workflows/CI-staging.yml:7]: hey @Sekhar-Kumar-Dash why did you change that?
that staging workflow is supposed to be run on PR to develop

the production-testing workflow runs on PR to master
is there a reason why you changed that?

Comment (Sekhar-Kumar-Dash) [reply .github/workflows/CI-staging.yml]: Hey @AlyaGomaa  sorry I removed it mistakenly. Thank you for correcting this mistake.

**Comment by AlyaGomaa** [comment]
Hello @Sekhar-Kumar-Dash do know why coverage fails here? 
https://github.com/stratosphereips/StratosphereLinuxIPS/actions/runs/8723300206/job/23931203964?pr=518

**Comment by Sekhar-Kumar-Dash** [comment]
hey @AlyaGomaa  sorry there was slight typo in whitelist path for coverage thats the reason why the test failed again

**Comment by Sekhar-Kumar-Dash** [comment]
Hey @AlyaGomaa , please accept my apologies for the recurring mistakes. This time, I have fixed it properly. I hope you don't mind

**Comment by Sekhar-Kumar-Dash** [comment]
There was a small typo in the vertical portscan test. It should have been "vertical portscans" and not "vertical portscan".
Sorry
